app/controllers/cart_controller.rb

```diff
@@ -7,7 +7,11 @@ def add_item
   end
 
   def show
-    @items = cart.items
+    if session[:user_id] && User.find(session[:user_id]).role == "admin"
+      render file: "/public/404"
+    else
+      @items = cart.items
+    end
```

Comment:
Very nice!  Don't we have a new helper method called current_user that we can use here?  Or am I getting my wires crossed?  I think we could simplify this on line 10 to `session[:user_id] && current_user == "admin"`